external/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala

```diff
@@ -42,10 +42,6 @@ object AvroUtils extends Logging {
       options: Map[String, String],
       files: Seq[FileStatus]): Option[StructType] = {
     val conf = spark.sessionState.newHadoopConf()
-    if (options.contains("ignoreExtension")) {
```

Comment:
@MaxGekk, let's just remove this if right after branch-3.0 is cut out.

Reply:
Shouldn't it be deprecated explicitly for users before removing? It should be mentioned in docs at least if I don't like to output a warning like in the PR.

Reply:
I think it still shows the warning properly although it only shows during schema inference. Yeah, can you simply fix the doc and say it's deprecated at `docs/sql-data-sources-avro.md`?

Reply:
+1 with @HyukjinKwon 
Let's remove the option and document it in the future, instead of creating such changes. If we merge this one, then there might be some other options we have to do the same thing.

Reply:
Here is the PR https://github.com/apache/spark/pull/27194 for docs